### PR TITLE
Fix clojure lsp commands not shown

### LINKED ISSUE #1610: Calva Refactor commands not appearing in command palette after initial start
In 2.0.256 the Calva Refactor commands don't show up in the command palette (I'm not sure if they're being registered), even after clojure-lsp has started. In 2.0.255, they do show up. CC @PEZ

### PR DESCRIPTION
**Pull request by bpringe**
## What has Changed?

- Clojure-lsp commands (like the refactor commands) now appear in the command palette after initial startup of the extension + clojure-lsp

Fixes #1610

## My Calva PR Checklist

I have:

- [x] Read [How to Contribute](https://github.com/BetterThanTomorrow/calva/wiki/How-to-Contribute#before-sending-pull-requests).
- [x] Directed this pull request at the `dev` branch. (Or have specific reasons to target some other branch.)
- [x] Made sure I have changed the PR base branch, so that it is not `published`. (Sorry for the nagging.)
- [x] Updated the `[Unreleased]` entry in `CHANGELOG.md`, linking the issue(s) that the PR is addressing.
- [x] Figured if **anything** about the fix warrants tests on Mac/Linux/Windows/Remote/Whatever, and either tested it there if so, or mentioned it in the PR.
- [x] Added to or updated docs in this branch, if appropriate
- [x] Tests
     - [x] Tested the particular change
     - [x] Figured if the change might have some side effects and tested those as well.
     - [x] Smoke tested the extension as such.
     - [ ] Tested the VSIX built from the PR (so, after you've submitted the PR). You'll find the artifacts by clicking _Show all checks_ in the CI section of the PR page, and then _Details_ on the `ci/circleci: build` test.
- [x] Referenced the issue I am fixing/addressing _in a commit message for the pull request_.
     - [x] If I am fixing the issue, I have used [GitHub's fixes/closes syntax](https://help.github.com/en/articles/closing-issues-using-keywords)
     - [x] If I am fixing just part of the issue, I have just referenced it w/o any of the "fixes” keywords.
- [x] Created the issue I am fixing/addressing, if it was not present.
- [x] Formatted all JavaScript and TypeScript code that was changed. (use the [prettier extension](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode) or run `npm run prettier-format`)
- [x] Confirmed that there are no linter warnings or errors (use the [eslint extension](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint), run `npm run eslint` before creating your PR, or run `npm run eslint-watch` to eslint as you go).

Ping @pez, @bpringe

### CODE REVIEW

**Comment by bpringe** [comment]
So… the fix really was just removing this one line: https://github.com/BetterThanTomorrow/calva/pull/1612/files#diff-f0c94fe155476e51d3ce1990a948fceb3394ba26272ac40eb49cfac2fbd93710L574

The code was setting a context that controls the visibility of the commands to false after activation. It was probably meant to be `true` , but there’s also a call to set it to `true` in `startClient`, so I removed the line from `activate`.

I also refactored the code in `startClient` to make it a bit easier to read and maybe a bit less error-prone (to/for me, at least). The main change is using a `try/catch` when awaiting `onReadyPromise`, rather than using `.catch`.

File: src/lsp/main.ts [hunk 1]
```diff
@@ -377,65 +409,30 @@ async function startClient(): Promise<boolean> {
...
+      testTreeHandler(tree);
+    });
+    client.onRequest('window/showMessageRequest', handleShowMessageRequest);
     await vscode.commands.executeCommand('setContext', 'clojureLsp:active', true);
     updateStatusItem('started');
-  } else {
+    return true;
+  } catch (error) {
+    console.error('clojure-lsp:', error);
+    setStateValue(LSP_CLIENT_KEY_ERROR, error);
```
Comment (PEZ) [inline src/lsp/main.ts:429]: Did you try this with a non-folder scenario, like with the Getting Started REPL? I ran into problems doing serverInfo there.

**Review by PEZ — COMMENTED** [review]
Looks like good changes! The scenarios I see we need to test are:

1. clojure-lsp initializes correctly if on auto-start
1. clojure-lsp initializes correctly if not auto-started, but started later
1. clojure-lsp is in a good state if it is stopped, then started again

And these should work when:

1. There is a folder open in the VS Code Window
1. There is not folder open

I'll try to test this during the day. If it is already tested to work. Please merge! 😄

**Comment by PEZ** [comment]
Tested the above now. Works like a charm! 🎉 

I take all your checked boxes as you are OK to merge, @bpringe . So doing that now.

**Pull request merged**